test/test_main.py

```diff
@@ -408,10 +408,16 @@ def _simulate_event(self, event_spec: EventSpec):
             if remote_app is not None:
                 env['JUJU_REMOTE_APP'] = remote_app
 
-            remote_unit = event_spec.remote_unit
-            if remote_unit is None:
-                remote_unit = ''
-            env['JUJU_REMOTE_UNIT'] = remote_unit
+            if (remote_app is None
+                or issubclass(event_spec.event_type, ops.RelationJoinedEvent)
+                or issubclass(event_spec.event_type, ops.RelationChangedEvent)
+                    or issubclass(event_spec.event_type, ops.RelationDepartedEvent)):
```

Comment:
Can you explain (perhaps in a comment) what this `if` block is testing for? I don't really understand the logic here or why it's needed.

Style nit: I'd probably indent all three `or`s the same (to the more-indented level of the last one).

Reply:
> Style nit: I'd probably indent all three `or`s the same (to the more-indented level of the last one).

This is where I do much prefer:

```python
if (
    remote_app is None
    or issubclass(event_spec.event_type, ops.RelationJoinedEvent)
    or issubclass(event_spec.event_type, ops.RelationChangedEvent)
    or issubclass(event_spec.event_type, ops.RelationDepartedEvent)
):
```

(with or without the first condition on a separate line).

But that doesn't match our formatting, so we have to have something that distinguishes the indent from the start of the actual block. Moved the others in.

Reply:
> Can you explain (perhaps in a comment) what this `if` block is testing for? I don't really understand the logic here or why it's needed.

Turns out that it is not needed :disappointed:. Earlier, when I had the `__init__` signatures changing, these tests were failing, and I thought I had tracked that down to this code passing the envvar when it shouldn't, and so this seemed more like the Juju behaviour. However, that really ought to just be done with changes in the event spec definitions I guess, and it also isn't needed any more.